bitrise.yml

```diff
@@ -3,165 +3,176 @@ default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
-  - APPLE_ID: $APPLE_ID
-  - APP_SPECIFIC_PASSWORD: $APP_SPECIFIC_PASSWORD
+    # prepare-test workflow params
+    - TEST_PROJECT_URL: https://github.com/bitrise-samples/Application-Loader-Test.git
+    - TEST_PROJECT_BRANCH: master
+    - BITRISE_CERTIFICATE_URL: $BITRISE_CERTIFICATE_URL
+    - BITRISE_CERTIFICATE_PASSPHRASE: $BITRISE_CERTIFICATE_PASSPHRASE
+    - INFO_PLIST_PATH: "./Application Loader Test/Info.plist"
+    - BITRISE_PROJECT_PATH: "./Application Loader Test.xcodeproj"
+    - BITRISE_SCHEME: Application Loader Test
+    # test workflow params
+    - API_KEY_PATH: $API_KEY_PATH
+    - API_KEY_ISSUER: $API_KEY_ISSUER
+    - APPLE_ID: $APPLE_ID
+    - APP_SPECIFIC_PASSWORD: $APP_SPECIFIC_PASSWORD
 
 workflows:
   ci:
     before_run:
-    - audit-this-step
-    - go-tests
+      - audit-this-step
+      - go-tests
     after_run:
-    - test
+      - test
 
   audit-this-step:
     title: Audit step.yml
     steps:
-    - script:
-        title: stepman audit --step-yml ./step.yml
-        inputs:
-        - content: |-
-            #!/bin/bash
-            set -ex
-            stepman audit --step-yml ./step.yml
+      - script:
+          title: stepman audit --step-yml ./step.yml
+          inputs:
+            - content: |-
+                #!/bin/bash
+                set -ex
+                stepman audit --step-yml ./step.yml
   go-tests:
     title: Run Go tests
     steps:
-    - go-list:
-    - golint:
-    - errcheck:
-    - go-test:
+      - go-list:
+      - golint:
+      - errcheck:
+      - go-test:
 
   test:
     title: Run E2E tests
     before_run:
-    - test-apple-id-connection-auth
-    - test-api-key-connection-auth
-    - test-apple-id-input-auth
-    - test-api-key-input-auth
+      - test-apple-id-connection-auth
+      - test-api-key-connection-auth
+      - test-apple-id-input-auth
+      - test-apple-id-connection-auth-withouth-password-input
+      - test-api-key-input-auth
 
   test-api-key-connection-auth:
     title: Test API key connection authentication
     before_run:
       - prepare-test
     steps:
-    - path::./:
-        inputs:
-        - connection: automatic
-        - altool_options: "--output-format xml --verbose"
+      - path::./:
+          inputs:
+            - connection: automatic
+            - altool_options: "--output-format xml --verbose"
 
   test-apple-id-connection-auth:
     title: Test Apple ID connection authentication
     before_run:
-    - prepare-test
+      - prepare-test
     steps:
-    - path::./:
-        inputs:
-        - connection: apple_id
-        - altool_options: "--output-format xml --verbose"
-        - app_password: $APP_SPECIFIC_PASSWORD # App-specific password can not be set for the connection
+      - path::./:
+          inputs:
+            - connection: apple_id
+            - altool_options: "--output-format xml --verbose"
+            - app_password: $APP_SPECIFIC_PASSWORD # App-specific password can not be set for the connection
```

Comment:
The comment is no longer true.
Despite this test, the value of APP_SPECIFIC_PASSWORD will not be used as the connection contains it, so I rather remove this test than for it to not really test out the input.

Reply:
There is a way to test this right. Can use the BITRISE_BUILD_URL with a file scheme to add a custom json file. BITRISE_BUILD_API_TOKEN can contain anything.
In a script step, download the connection json, use jq to remove the application specific password and write to a file.

Reply:
I agree with the comment remove part, but I would keep the test to cover the case of having the `app_password` still set but the actual password being used from the connection. Basically to cover regression.

Reply:
In that case let's use a known invalid app specific password. Do not totally agree with the regression part as the positive case will not be tested.

Reply:
I agree with Laci, and I would avoid adding more e2e test cases if possible (daily deploy limit issues).
I think the unit tests are covering the app-specific password selection.

Reply:
Ok, i'll remove this test then 👍